src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -343,19 +353,48 @@ def main(args):
         try:
             upload_verified_config(config_loader,
                                    local_store,
-                                   args.config_type,
+                                   selected_config,
                                    args.force,
                                    args.autoconfirm)
         except (UserAbort, ConfigUploadFailed) as exc:
             log.error("Upload failed")
             sys.exit(exc)
 
 
+def lookup_config_type(config_type, config_location):
+    """
+    This takes the string defining the config_type from arg.parse and returns
+    a instance of the class containing data on the config type selected
+    """
+    config_classes = load_plugins_in_dir(PLUGIN_DIR, config_location)
+    for choice in config_classes:
+        if config_type == choice.name:
+            return choice
+        # The config_type will always be the choice.name of one class as
+        # parse_arguments will make sure config_type == choice.name
+
+
 def parse_arguments():
     """
     Parse the arguments passed to the script.
+    The options for config type are designated by the plugins available
     :return:
     """
+    # Pick the options to appear in argument parsing, the config types are
+    # loaded as plugin classes onto the node so only the config files available
+    # on the node are able to be changed on the node.
+    config_classes = load_plugins_in_dir(PLUGIN_DIR)
+
+    # If there are no config_classes loaded from the PLUGIN_DIR this means you
+    # will be unable to select anything, hence you want to raise an error, that
+    # makes sense. Otherwise you would get always get refused by the arg
+    # parsing.
+    if config_classes == []:
+        raise NoConfigFound
```

Comment:
Be careful when raising exceptions. You should only do it when one of the following is true:
- You have handling code that will catch the exception somewhere.
- You are comfortable with the exception being shown to whoever is running your script.

I don't think either of these two is true here at the moment. If you hit this exception, the user will be shown a Python stack trace - it probably won't look very nice!

But the question remains - how should we handle this circumstance? Have a look and see if you can figure out how we do this elsewhere in the script, then come and have a chat with me about it and we can go through it together.